charts/kong/templates/_helpers.tpl

```diff
@@ -454,7 +454,11 @@ The name of the service used for the ingress controller's validation webhook
 
 {{- $autoEnv := dict -}}
   {{- $_ := set $autoEnv "CONTROLLER_KONG_ADMIN_TLS_SKIP_VERIFY" true -}}
-  {{- $_ := set $autoEnv "CONTROLLER_PUBLISH_SERVICE" (printf "%s/%s" ( include "kong.namespace" . ) ( .Values.proxy.nameOverride | default ( printf "%s-proxy" (include "kong.fullname" . )))) -}}
+  {{- if semverCompare "< 3.0.0" (include "kong.effectiveVersion" .Values.ingressController.image) -}}
+    {{- $_ := set $autoEnv "CONTROLLER_PUBLISH_SERVICE" (printf "%s/%s" ( include "kong.namespace" . ) ( .Values.proxy.nameOverride | default ( printf "%s-proxy" (include "kong.fullname" . )))) -}}
+  {{- else -}}
+    {{- $_ := set $autoEnv "CONTROLLER_INGRESS_SERVICE" (printf "%s/%s" ( include "kong.namespace" . ) ( .Values.proxy.nameOverride | default ( printf "%s-proxy" (include "kong.fullname" . )))) -}}
+  {{- end -}}
```

Comment:
Wouldn't it be better to set the variable name to `CONTROLLER_PUBLISH_SERVICE` or `CONTROLLER_INGRESS_SERVICE` based on the version and only have 1 block setting the envvar?